cxx-sensors/src/main/java/org/sonar/cxx/sensors/clangsa/CxxClangSASensor.java

```diff
@@ -88,7 +90,12 @@ protected void processReport(final SensorContext context, File report)
         NSString desc = (NSString) diag.get("description");
         String description = desc.getContent();
 
-        String checkerName = ((NSString) diag.get("check_name")).getContent();
+        final NSObject checkNameValue = diag.get("check_name");
+        if(checkNameValue == null) {
+            LOG.error("clang version > 3.7.0 is required. Your version is: " + clangVersion);

```

Comment:
Would remove this error, see comment above

Reply:
Checking for presence of `check_name` allows to parse output of custom-named internal builds of clang like `prefix-version-suffix`. I'm ok to check the version, this makes situation clearer.

However you should make this decision: support only names with numbers in front or check for existence of `check_name` instead.